### PR TITLE
DM-1651: New command listener

### PR DESCRIPTION
**Pull request by SkyToGround**
### Description of work

* Moves and modifies the command listener.
* Suppresses unusedFunction-warnings to simplify use of small PRs.

### CODE REVIEW

File: src/CommandSystem/CommandListener.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+using FileWriter::Msg;
+
+/// \brief Check for new commands on the topic.
+class CommandListener {
+public:
+  CommandListener(uri::URI CommandTopicUri, Kafka::BrokerSettings Settings);
+
+  /// \brief Poll the Kafka topic for a new command.
+  /// \note Will timeout on its fist call. Will continue to timeout as long as
```
Comment (SkyToGround) [inline src/CommandSystem/CommandListener.h:27]:
```suggestion
  /// \note Will timeout on its first call. Will continue to timeout as long as
```

File: src/tests/CMakeLists.txt [hunk 2]
```diff
@@ -41,7 +41,7 @@ set(UnitTests_SRC
         $<TARGET_OBJECTS:NeXusDatasetTests>
         StatusReporterTests.cpp
         helpers/StatusHelpers.cpp
-        HelperTests.cpp)
+        HelperTests.cpp CommandSystem/CommandListenerTests.cpp)
```
Comment (matthew-d-jones) [inline src/tests/CMakeLists.txt:44]: Missing new line

**Review by matthew-d-jones — COMMENTED** [review]
The approach to retry the metadata call on poll looks good to me :+1: 

I see an error message, and not the expected warning message, during the unit test
```
[15:25:02.594365] [error] [processID: 45805]: error at type: -1
```
Any idea what is going on there?

It would be nicer to pass the consumer to the command listener, so that you can pass a fake or mock and also test the happy path.

File: src/tests/CMakeLists.txt (hunk 2 above)
Comment (SkyToGround) [reply src/tests/CMakeLists.txt:44]:
```suggestion
        HelperTests.cpp CommandSystem/CommandListenerTests.cpp
        )
```

**Comment by SkyToGround** [comment]
@matthew-d-jones We do get the expected warning and error messages based on the configured log level.
As for providing an external consumer. This is one of the cases where my guesstimate is that we need on the order of 100 lines of more code to enable that particular unit test. That together with the consequence and probability of failure in this class (low and low) makes me think that this is not worth it. Still, I can do it but I want to hear the opinion of someone else in that case as well.

**Review by matthew-d-jones — APPROVED** [review]
Sorry, I meant pass "consumer factory", I see that not constructing the consumer in the listener would require more work. If that doesn't change matters then I leave it to your discretion.

**Pull request merged**